Running `testing-in-the-modular-world` in direct mode fails

I'm facing strange issue. Fortunately it can be easily reproduced in the https://github.com/sormuras/testing-in-the-modular-world.git project.

Steps to reproduce:
1. set `junit-jupiter-api` version to 5.9.1
2. set `maven-surefire-plugin` version to 3.0.0-M7
3. set `junit-platform-maven-plugin` version to 1.1.6
4. change `executor` to `DIRECT`

Expected:
- The build passes

Actual:

```
/mnt/nvme/opt/java/jdk/bin/java -Dmaven.multiModuleProjectDirectory=/mnt/hdd/home/opc/Projects/testing-in-the-modular-world -Dmaven.home=/mnt/nvme/opt/java-tools/apache-maven-3.8.6 -Dclassworlds.conf=/mnt/nvme/opt/java-tools/apache-maven-3.8.6/bin/m2.conf -Dmaven.ext.class.path=/mnt/nvme/opt/jetbrains/apps/IDEA-U/ch-0/222.4167.29/plugins/maven/lib/maven-event-listener.jar -javaagent:/mnt/nvme/opt/jetbrains/apps/IDEA-U/ch-0/222.4167.29/lib/idea_rt.jar=39069:/mnt/nvme/opt/jetbrains/apps/IDEA-U/ch-0/222.4167.29/bin -Dfile.encoding=UTF-8 -classpath /mnt/nvme/opt/java-tools/apache-maven-3.8.6/boot/plexus-classworlds.license:/mnt/nvme/opt/java-tools/apache-maven-3.8.6/boot/plexus-classworlds-2.6.0.jar org.codehaus.classworlds.Launcher -Didea.version=2022.2.2 install
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------< de.sormuras:testing-in-the-modular-world >--------------
[INFO] Building testing-in-the-modular-world 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ testing-in-the-modular-world ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /mnt/hdd/home/opc/Projects/testing-in-the-modular-world/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ testing-in-the-modular-world ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 5 source files to /mnt/hdd/home/opc/Projects/testing-in-the-modular-world/target/classes
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ testing-in-the-modular-world ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ testing-in-the-modular-world ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2 source files to /mnt/hdd/home/opc/Projects/testing-in-the-modular-world/target/test-classes
[INFO] 
[INFO] --- junit-platform-maven-plugin:1.1.6:launch (injected-launch) @ testing-in-the-modular-world ---
[INFO] Launching JUnit Platform 1.9.1...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.841 s
[INFO] Finished at: 2022-09-28T13:47:52+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal de.sormuras.junit:junit-platform-maven-plugin:1.1.6:launch (injected-launch) on project testing-in-the-modular-world: Execution failed!: java.lang.ClassNotFoundException: de.sormuras.junit.platform.isolator.worker.Worker -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

Process finished with exit code 1
```

Relevant debug log:
```
[INFO] --- junit-platform-maven-plugin:1.1.6:launch (injected-launch) @ testing-in-the-modular-world ---
[DEBUG] Configuring mojo de.sormuras.junit:junit-platform-maven-plugin:1.1.6:launch from plugin realm ClassRealm[extension>de.sormuras.junit:junit-platform-maven-plugin:1.1.6, parent: jdk.internal.loader.ClassLoaders$AppClassLoader@251a69d7]
[DEBUG] Configuring mojo 'de.sormuras.junit:junit-platform-maven-plugin:1.1.6:launch' with basic configurator -->
[DEBUG]   (f) dryRun = false
[DEBUG]   (f) execution = de.sormuras.junit:junit-platform-maven-plugin:1.1.6:launch {execution: injected-launch}
[DEBUG]   (f) executionProgress = 60
[DEBUG]   (f) executor = DIRECT
[DEBUG]   (f) isolation = NONE
[DEBUG]   (f) mavenBuild = org.apache.maven.model.Build@3d8b319e
[DEBUG]   (f) mavenProject = MavenProject: de.sormuras:testing-in-the-modular-world:1.0-SNAPSHOT @ /mnt/hdd/home/opc/Projects/testing-in-the-modular-world/pom.xml
[DEBUG]   (f) mavenRepositorySession = org.eclipse.aether.DefaultRepositorySystemSession@27a97e08
[DEBUG]   (f) mavenSession = org.apache.maven.execution.MavenSession@5b12012e
[DEBUG]   (f) skip = false
[DEBUG]   (f) tags = []
[DEBUG]   (f) targetDirectory = /mnt/hdd/home/opc/Projects/testing-in-the-modular-world/target/junit-platform
[DEBUG]   (f) timeout = 300
[DEBUG] -- end configuration --
[DEBUG] Executing JUnitPlatformMojo...
[INFO] Launching JUnit Platform 1.9.1...
[DEBUG] Path
[DEBUG]   java.home = /mnt/nvme/opt/java/jdk-17.0.4.1
[DEBUG]   user.dir = /mnt/hdd/home/opc/Projects/testing-in-the-modular-world
[DEBUG]   project.basedir = /mnt/hdd/home/opc/Projects/testing-in-the-modular-world
[DEBUG] Class Loader
[DEBUG]   mojo's loader = ClassRealm[extension>de.sormuras.junit:junit-platform-maven-plugin:1.1.6, parent: jdk.internal.loader.ClassLoaders$AppClassLoader@251a69d7]
[DEBUG]   context loader = ClassRealm[extension>de.sormuras.junit:junit-platform-maven-plugin:1.1.6, parent: jdk.internal.loader.ClassLoaders$AppClassLoader@251a69d7]
[DEBUG]   platform loader = jdk.internal.loader.ClassLoaders$PlatformClassLoader@2009f9b0
[DEBUG] Artifact Map
[DEBUG]   org.apiguardian:apiguardian-api                    -> org.apiguardian:apiguardian-api:jar:1.1.2:test trail=[de.sormuras:testing-in-the-modular-world:jar:1.0-SNAPSHOT, org.junit.jupiter:junit-jupiter-api:jar:5.9.1, org.apiguardian:apiguardian-api:jar:1.1.2]
[DEBUG]   org.junit.jupiter:junit-jupiter-api                -> org.junit.jupiter:junit-jupiter-api:jar:5.9.1:test trail=[de.sormuras:testing-in-the-modular-world:jar:1.0-SNAPSHOT, org.junit.jupiter:junit-jupiter-api:jar:5.9.1]
[DEBUG]   org.junit.platform:junit-platform-commons          -> org.junit.platform:junit-platform-commons:jar:1.9.1:test trail=[de.sormuras:testing-in-the-modular-world:jar:1.0-SNAPSHOT, org.junit.jupiter:junit-jupiter-api:jar:5.9.1, org.junit.platform:junit-platform-commons:jar:1.9.1]
[DEBUG]   org.opentest4j:opentest4j                          -> org.opentest4j:opentest4j:jar:1.2.0:test trail=[de.sormuras:testing-in-the-modular-world:jar:1.0-SNAPSHOT, org.junit.jupiter:junit-jupiter-api:jar:5.9.1, org.opentest4j:opentest4j:jar:1.2.0]
[DEBUG] Version
[DEBUG]   java.version = 17.0.4.1
[DEBUG]   java.class.version = 61.0
[DEBUG]   isolator.version = 1.0.0-M10
[DEBUG]   junit.platform.version = 1.9.1
[DEBUG]   junit.jupiter.version = 5.9.1
[DEBUG]   junit.vintage.version = 5.3.2
[DEBUG] Java Module System
[DEBUG]   main -> module foo { requires=[java.base] packages=[baz, foo] }
[DEBUG]   test -> <empty>
[DEBUG]   mode -> MODULAR_PATCHED_TEST_RUNTIME
[DEBUG]   module-info.test -> /mnt/hdd/home/opc/Projects/testing-in-the-modular-world/target/test-classes/module-info.test
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.486 s
[INFO] Finished at: 2022-09-28T13:53:34+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal de.sormuras.junit:junit-platform-maven-plugin:1.1.6:launch (injected-launch) on project testing-in-the-modular-world: Execution injected-launch of goal de.sormuras.junit:junit-platform-maven-plugin:1.1.6:launch failed: Conversion = s, Flags =   -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal de.sormuras.junit:junit-platform-maven-plugin:1.1.6:launch (injected-launch) on project testing-in-the-modular-world: Execution injected-launch of goal de.sormuras.junit:junit-platform-maven-plugin:1.1.6:launch failed: Conversion = s, Flags =  
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:375)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution injected-launch of goal de.sormuras.junit:junit-platform-maven-plugin:1.1.6:launch failed: Conversion = s, Flags =  
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: java.util.FormatFlagsConversionMismatchException: Conversion = s, Flags =  
    at java.util.Formatter$FormatSpecifier.failMismatch (Formatter.java:4438)
    at java.util.Formatter$FormatSpecifier.checkBadFlags (Formatter.java:3176)
    at java.util.Formatter$FormatSpecifier.checkGeneral (Formatter.java:3134)
    at java.util.Formatter$FormatSpecifier.<init> (Formatter.java:2896)
    at java.util.Formatter.parse (Formatter.java:2747)
    at java.util.Formatter.format (Formatter.java:2671)
    at java.util.Formatter.format (Formatter.java:2625)
    at java.lang.String.format (String.java:4143)
    at de.sormuras.junit.platform.maven.plugin.MavenDriver.lambda$copyDirectoryStructure$1 (MavenDriver.java:315)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept (ForEachOps.java:183)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:197)
    at java.util.Iterator.forEachRemaining (Iterator.java:133)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining (Spliterators.java:1845)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:509)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:499)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential (ForEachOps.java:150)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential (ForEachOps.java:173)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach (ReferencePipeline.java:596)
    at de.sormuras.junit.platform.maven.plugin.MavenDriver.copyDirectoryStructure (MavenDriver.java:303)
    at de.sormuras.junit.platform.maven.plugin.MavenDriver.buildPathMap (MavenDriver.java:138)
    at de.sormuras.junit.platform.maven.plugin.JUnitPlatformMojo.execute (JUnitPlatformMojo.java:384)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException

Process finished with exit code 1
```

Environment:
- Linux x86_64
- JDK 17
   ```
   java version "17.0.4.1" 2022-08-18 LTS
   Java(TM) SE Runtime Environment (build 17.0.4.1+1-LTS-2)
   Java HotSpot(TM) 64-Bit Server VM (build 17.0.4.1+1-LTS-2, mixed mode, sharing)
   ```
- Maven 3.8.6
   ```
   Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)
   Maven home: /mnt/nvme/opt/java-tools/maven
   Java version: 17.0.4.1, vendor: Oracle Corporation, runtime: /mnt/nvme/opt/java/jdk-17.0.4.1
   Default locale: en_US, platform encoding: UTF-8
   OS name: "linux", version: "5.15.0-2.52.3.el8uek.x86_64", arch: "amd64", family: "unix"
   ```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -16,7 +16,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        java: [18]
+        java: [19]
 
     steps:
       - name: 'Check out sources'

--- a/src/main/java/de/sormuras/junit/platform/maven/plugin/MavenDriver.java
+++ b/src/main/java/de/sormuras/junit/platform/maven/plugin/MavenDriver.java
@@ -39,6 +39,7 @@ import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.Comparator;
 import java.util.LinkedHashMap;
 import java.util.LinkedHashSet;
 import java.util.List;
@@ -135,6 +136,15 @@ class MavenDriver implements Driver {
       testPaths.add(patchedDirectory);
 
       try {
+        // Copying directory structure fails if destination is non-empty
+        // so we recursively delete the previous patchedDirectory if it exists
+        Path patchedDirPath = Paths.get(patchedDirectory);
+        if (Files.exists(patchedDirPath)) {
+          try (Stream<Path> pathStream = Files.walk(patchedDirPath)) {
+            // subdirectories will sort alphabetically after parents, so reverse order
+            pathStream.sorted(Comparator.reverseOrder()).forEach(p -> p.toFile().delete());
+          }
+        }
         copyDirectoryStructure(mainDirectory, patchedDirectory, false);
       } catch (IOException e) {
         throw new UncheckedIOException(

